src/Collateral.hs

```diff
@@ -146,7 +146,9 @@ mkValidator contractInfo@ContractInfo{..} dat _ ctx = validate
 
     checkForLiquidationToken :: Bool
     checkForLiquidationToken =
-      any (\(cs, _, n) -> cs == liquidateNft dat && n > 0) (U.mintFlattened ctx)
+      all (\(_, _, n) -> n > 0)
+      $ filter (\(cs, _, n) -> cs == liquidateNft dat)
+      (U.mintFlattened ctx)
```

Comment:
@tomazvila Careful, this change also means that you do not need a liquidation token to be minted anymore (imagine the `filter` returning an empty array). A transaction with no liquidation token minted or burned in the transaction passes the liquidation checks now.

Reply:
https://github.com/aadafinance/aada-finance/pull/48